### PR TITLE
Adds lightbox support for images

### PR DESCRIPTION
**Pull request by markdembo**
Solution based on:
https://hugocodex.org/add-ons/lightbox/
I thought for a moment, it's hugo official, sadly it's just community.

Automatically converts links to images to lightbox, if they contain an image link. Images without link will not be affected

More comments on the individual files.

### CODE REVIEW

File: content/kv/reference/how-kv-works.md [hunk 1]
```diff
@@ -6,27 +6,27 @@ weight: 7
...
+KV supports exceptionally high read volumes with low latency, making it possible to build highly dynamic APIs.
 
 While reads are periodically revalidated in the background, requests which are not in cache and need to hit the centralized back end can experience high latencies.
 
 ## Write data to KV and read data from KV
 
 When you write to KV, your data is written to central data stores. Your data is not sent automatically to every location’s cache, but regional tiers are notified within seconds to do a purge of that key.
 
-![Your data is written to central data stores when you write to KV.](/images/kv/kv-write.svg)
+[![Your data is written to central data stores when you write to KV.](/images/kv/kv-write.svg)](/images/kv/kv-write.svg)
```
Comment (markdembo) [inline content/kv/reference/how-kv-works.md:19]: Example usage. Image needs to be a link to image.

File: layouts/partials/footer.html [hunk 2]
```diff
@@ -67,4 +67,6 @@
       {{- end -}}
     </div>
   </div>
+  <script type="text/javascript" src="/lightbox.js"></script>
+  <link rel="stylesheet" href="/lightbox.css">
```
Comment (markdembo) [inline layouts/partials/footer.html]: Adds this script to all pages, if we want to be more conservative, this should be changed to a per-page level first. I'd then remove this includes here from the PR and then each now docs writer could include these lines in the individual PR

File: static/lightbox.css [hunk 3]
```diff
@@ -0,0 +1,156 @@
+#lightbox {
```
Comment (markdembo) [inline static/lightbox.css:1]: Nothing fancy here. Copy & Paste from original.

File: static/lightbox.js [hunk 4]
```diff
@@ -0,0 +1,163 @@
+function is_youtubelink(url) {
+  var p =
+    /^(?:https?:\/\/)?(?:www\.)?(?:youtu\.be\/|youtube\.com\/(?:embed\/|v\/|watch\?v=|watch\?.+&v=))((\w|-){11})(?:\S+)?$/;
+  return url.match(p) ? RegExp.$1 : false;
+}
+function is_imagelink(url) {
+  var p = /([a-z\-_0-9\/\:\.]*\.(jpg|jpeg|png|gif|svg|avif|webp))/i;
```
Comment (markdembo) [inline static/lightbox.js:2]: Modified original to included `svg`, `avif` and `webp`

File: layouts/partials/footer.html [hunk 5]
```diff
@@ -67,4 +67,6 @@
       {{- end -}}
     </div>
   </div>
+  <script type="text/javascript" src="/lightbox.js"></script>
```
Comment (kodster28) [inline layouts/partials/footer.html]:
```suggestion
  <script type="text/javascript" src="/lightbox.js" defer></script>
```

File: static/lightbox.js [hunk 6]
```diff
@@ -0,0 +1,163 @@
+function is_youtubelink(url) {
```
Comment (kodster28) [inline static/lightbox.js]: I don't _think_ this'll work with our current YT videos b/c they're styled as iframes + use the YT `nocookie` url.

Maybe worth updating this code to account for the nocookie URL + this [shortcode template](https://github.com/cloudflare/cloudflare-docs/blob/production/layouts/shortcodes/youtube.html) to make them render as links (or adjusting the lightbox.js)?

Example visible at https://developers.cloudflare.com/workers-ai/tutorials/how-to-choose-the-right-text-generation-model/

**Comment by markdembo** [comment]
Removed videos from scope

**Review by kodster28 — APPROVED** [review]
Question for the future.... is there a reason we _wouldn't_ want this to be the default for all images?

**Pull request merged**